metpy/cbook.py

```diff
@@ -39,4 +37,103 @@ def get_test_data(fname, as_file_obj=True):
+
+        # Check if there is a cache version file. If not, this is a new cache, so create it.
+        cache_version_file_path = os.path.join(self.cache_home, 'cache_version.txt')
+        if not os.path.exists(cache_version_file_path ):
+            with open(cache_version_file_path , 'w') as f:
+                f.write('{}\n'.format(self.cache_version))
+
+        # Check if this is an old cache version. If it is, remove the cache so we are sure
+        # to be using current files
+        with open(cache_version_file_path, 'r') as f:
+            current_cache_version = int(f.readline())
+        if current_cache_version != self.cache_version:
+            self.clear_cache()
+            # Now that we've blown the old cache away, create a new one and
+            # a version file with the current cache version
+            self.cache_home = self.get_cache_location()
+            with open(cache_version_file_path, 'w') as f:
+                f.write('{}\n'.format(self.cache_version))
+
+    def get_cache_location(self):
+        """Get the cache location."""
+
+        # Look for the staticdata directory (i.e. this is a git checkout)
+        if os.path.exists(os.path.join(os.path.dirname(__file__), '..', 'staticdata')):
+            cache_path = os.path.join(os.path.dirname(__file__), '..', 'staticdata')
+
+        # If there is no static data directory, use the OS default
+        else:
+            # Get the cache location for this OS
+            cache_path = self.get_os_cache_path()
+
+        # Look for a user-defined cache location - this takes priority over anything else
+        if os.environ.get('METPY_CACHE_LOCATION'):
+            cache_path = os.environ.get('METPY_CACHE_LOCATION')
+
+        # Make sure we expand ~ to the user home
+        cache_path = os.path.expanduser(cache_path)
+
+        # Make sure there is a cache directory there. If not, let's create it so we have
+        # somewhere to put the data.
+        if not os.path.exists(cache_path):
+            os.makedirs(cache_path)
+
+        return cache_path
+
+    def get_os_cache_path(self):
+        platform = sys.platform
+
+        if platform == 'darwin':
+            cache_path = '~/Library/Caches/MetPy'
+        elif platform == 'win32':
+            cache_path = '~/AppData/Local/MetPy/cache'
+        else:  # *NIX
+            cache_path = '~/.cache/MetPy'
+
+        return os.path.expanduser(cache_path)
+
+    def get_data(self, fname, as_file_obj=True):
+        """Get data from the cache."""
+
+        file_path = os.path.join(self.cache_home, fname)
+
+        # See if we already have the file locally
+        if os.path.exists(file_path):
+            return file_path
+
+        # If we don't have it, try to download it
+        else:
+            self.download_data(fname)
+
+        return file_path
+
+    def download_data(self, fname):
+        """Download remote data to the cache."""
+        if '.dirty' in __version__:
```

Comment:
dirty means that there are uncommitted changes, so what you really want to look at is whether there's anything beyond the e.g. `0.7.0`. Maybe: `len(__version__.split('+')) > 1` ? `_version` module has `get_versions`, but whether it's a git snapshot isn't in there.

Reply:
What about just checking for `+` in the version string?

Reply:
That seems better.